src/ILCompiler.Compiler/src/IL/Stubs/Bootstrap/BootstrapMain.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Internal.TypeSystem;
+using Internal.IL;
+using Internal.IL.Stubs;
+using ILCompiler.Compiler.IL.Stubs;
+using System;
+
+namespace Internal.IL.Stubs.Bootstrap
+{
+    /// <summary>
+    /// Bootstrap code that does initialization, Main invocation
+    /// and shutdown of the runtime.
+    /// </summary>
+    public sealed class BootstrapMainMethod : BootstrapMethod
+    {
+        private MethodDesc _mainMethod;
+
+        public BootstrapMainMethod(DefType owningType, MethodDesc mainMethod)
+            : base(owningType)
+        {
+            _mainMethod = mainMethod;
+        }
+
+        public override string Name
+        {
+            get
+            {
+                return "Internal_IL_Stubs_BootstrapMain";
+            }
+        }
+
+        public override MethodIL EmitIL()
+        {
+            ILEmitter emitter = new ILEmitter();
+            ILCodeStream codeStream = emitter.NewCodeStream();
+            ILLocalVariable returnValue = emitter.NewLocal(Context.GetWellKnownType(WellKnownType.Int32));
+
+            TypeDesc bootstrap = Context.SystemModule.GetType("System.Runtime", "BootstrapHelpers");
+
+            codeStream.Emit(ILOpcode.call, emitter.NewToken(bootstrap.GetMethod("Initialize", null)));
+            if (_mainMethod.Signature.Length > 0)
+            {
+                MethodDesc getArgs = bootstrap.GetMethod("GetCommandLineArgs", null);
```

Comment:
The GetCommandLineArgs method should be called all the time, even the main method does not take is as argument. Maybe a better name for the method would be `InitializeCommandLineArgs`.

We will need it to implement `Environment.GetCommandLineArgs()`. It was added recently to the [contract](https://github.com/dotnet/corefx/blob/master/src/System.Runtime.Extensions/ref/System.Runtime.Extensions.cs#L448) and was not implemented for CoreRT yet. It would be great if you would could implement `Environment.GetCommandLineArgs()` as part of this change (it should be just a few lines of extra code - cache the command line args in static and return a clone).